.github/workflows/publish.yml

```diff
@@ -1,7 +1,11 @@
-name: Pypi
+name: Publish to PyPI
 
 on:
-  push:
+  workflow_run:
+    workflows: ["Test"]
+    branches: [main]
+    types:
+      - completed
     tags:
     - 'v*'
```

Comment:
The test workflow will not be run on tags, it's only on pushed events. So, I guess this will never be triggered.
You can leave it as is.